src/vmm/src/vmm_config/entropy.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+use devices::virtio::rng::{Entropy, Error as EntropyError};
+use serde::{Deserialize, Serialize};
+
+/// This struct represents the strongly typed equivalent of the json body from entropy device
+/// related requests.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, Default)]
+#[serde(deny_unknown_fields)]
+pub struct EntropyDeviceConfig {}
+
+impl From<&Entropy> for EntropyDeviceConfig {
+    fn from(_dev: &Entropy) -> Self {
+        EntropyDeviceConfig::default()
+    }
+}
+
+/// Errors that can occur while handling configuration for
+/// an entropy device
+#[derive(Debug, thiserror::Error)]
+pub enum EntropyDeviceError {
+    /// Error while create an entropy device
+    #[error("Could not create Entropy device: {0}")]
+    CreateDevice(#[from] EntropyError),
+}
+
+type Result<T> = std::result::Result<T, EntropyDeviceError>;
+
+/// A builder type used to construct an Entropy device
+#[derive(Default)]
+pub struct EntropyDeviceBuilder(Option<Arc<Mutex<Entropy>>>);
+
+impl EntropyDeviceBuilder {
+    /// Create a new instance for the builder
+    pub fn new() -> Self {
+        Self(None)
+    }
```

Comment:
This is the same as `EntropyDeviceBuilder::default()` would it make sense to remove this duplicated code?

Reply:
As Patrick mentioned in one of the previous comments, I think that all of this infrastructure that parses and creates devices from config needs re-work, so I'd rather follow the same pattern as other devices and just fix all of this in a later PR. What do you think?